File: Programmers/Level2/N개의최소공배수/N개의최소공배수.js
function solution(arr) {
    const gcd = (a, b) => (!b ? a : gcd(b, a % b));
    const lcm = (a, b) => (a * b) / gcd(a, b);

    arr.sort((a, b) => b - a);
    let answer = arr[0];

    for (let i = 1; i < arr.length - 1; i++) {
        answer = lcm(answer, arr[i]);
    }
    return answer;
}

File: Programmers/Level2/N개의최소공배수/N개의최소공배수.py
def solution(arr):
    gcd = lambda a, b : a if not b else gcd(b, a % b)
    lcm = lambda a, b : a * b // gcd(a, b)
    
    arr.sort(reverse=True)
    answer = arr[0]
    
    for i in range(len(arr)):
        answer = lcm(answer, arr[i])
        
    return answer